### ISSUE TITLE
Error response from daemon: driver failed programming external connectivity on endpoint

### ISSUE THREAD

**Issue by gitULuser** [docker/for-win#2722]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:
082AAF50-9E4F-494D-A533-A10EFE9F0A05/20181009115433
### Expected behavior
Docker container runs.
### Actual behavior
Getting an error message when issuing the run command:
"Program Files\Docker\Docker\Resources\bin\docker.exe: 
Error response from daemon: driver failed programming external connectivity on endpoint test01 (3d48308d320a9c42b0d6583bb1add664f975cbb58f26c4ccb805a0287b019f66): 
Error starting userland proxy: mkdir /port/tcp:0.0.0.0:8080:tcp:172.17.0.2:80: input/output error."
### Information

It is reproducible, but the error message is only displayed the first time.
After the initial time, there is no error message, but the container does not run. Details are shown as CREATED: 7 seconds ago STATUS: Exited (1) 6 seconds ago.
Tried restarting Docker, and also resetting to factory defaults.

  - Windows Version: Windows 10 Enterprise 1803 build 17134.286
  - Docker for Windows Version: 18.06.1-ce

### Steps to reproduce the behavior

  1. Create the container.
  2.  Run the container.
  3.  Check to see if it is running.

**Comment by jasonwilliams** [comment]
This is the same for me, its always the first time.
After i restart docker the second time it works

**Comment by jasonwilliams** [comment]
Properly shutting docker down before restart worked for me

**Comment by pd93** [comment]
Also related: https://github.com/docker/for-win/issues/1967

**Comment by thomasvnl** [comment]
/remove-lifecycle stale

> This is the same for me, its always the first time.
> After i restart docker the second time it works

Same issue, also on Windows 10 Pro. After restarting docker, it works again.

**Comment by lbs-rodrigo** [comment]
I also restarted the docker on windows 10 and it worked.

**Comment by xenu256** [comment]
Yeah, restarting work. Docker version 18.09.1, build 4c52b90

**Comment by zanfilip** [comment]
I'd really like to see a solution other than restarting docker - which is disruptive. It would be preferable to have a command to release the process holding the lock on the endpoint _despite the container being stopped_. Is that possible?

**Comment by pd93** [comment]
@zanfilip This is caused by an incompatibility with Docker and fastboot. You can either make sure you stop all containers before shutting Windows down or you can disable fastboot in Windows' power settings by doing the following:

`CTRL+R > "powercfg.cpl" > "Choose what the power buttons do" > "Change settings that are currently unavailable" > Deselect "Turn on fast start-up"`

You can also disable fastboot with a single command in powershell if you're comfortable doing so:

`Set-ItemProperty 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Power\' -Name HiberbootEnabled -Value 0`

More info in #1967 as referenced earlier.

**Comment by FrankFang** [comment]
@pd93 I prefer to restart docker instead of disabled fastboot. Because fastboot is cool : )

**Comment by zanfilip** [comment]
@pd93 Thanks for the informative suggestion but **I have the issue each time I stop a container**, not after restarting Windows.

I have to restart that container because it exposes 2 ports and often within minutes of being up the ports are no longer accessible (this is ibmcom/mq latest, in case anyone is interested). So the following commands give me the issue:
`CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                                                      NAMES
69dfe4b24676        ibmcom/mq           "runmqdevserver"    8 days ago          Up About a minute   0.0.0.0:1414->1414/tcp, 0.0.0.0:9443->9443/tcp, 9157/tcp   pensive_mcnulty`
`docker container stop pensive_mcnulty`
`docker start -ai pensive_mcnulty`

**Comment by LBoraz** [comment]
same issue on windows 10

**Comment by hannahbaker28** [comment]
Well this does my head in.. every morning I turn my computer on, run my build command only to read :

Error response from daemon: driver failed programming external connectivity on endpoint sharp_lichterman (e88c95d64534c769171ed780db3ece823eafb1ddd5616c58d7334f766a8cd9c0): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:1533:tcp:172.17.0.2:1433: input/output error.

I shouldn't have to restart docker every morning and wait 10 years for it to come back up...
It's supposed to be an 'in' technology but man this is not efficient.

I'll be waiting for your fix! (ASAP I hope!)

**Comment by TimGoeller** [comment]
Got the same problem right now, although it didn't happen the first time I restarted the machine, it started to show these problems after ~ a week.

**Comment by Touseef90** [comment]
Facing the same issue, at the first time. After I restart docker it works fine.
I was using this command "docker run --detach --publish 80:80 --name webserver nginx". On windows OS.

**Comment by LordGenry** [comment]
Got the same issue. 
```bash
docker run -p 3001:3001 -d $NAME
```
Solutions as:
- disable fastboot
- restarting docker
-> are not really solutions, it's a duck-tape

**Comment by sentinel09** [comment]
Started to experience this issue.
Worked great 1-1.5 week ago.

WIndows 10 Pro 17763
Docker:  18.09.2

**Comment by dragonworx** [comment]
I found restarting docker did fix it, but it was REALLY slow - too painful for any kind of enjoyable development...
I narrowed down the problem to containers which were killed by the user (Ctrl-C), they left running container processes around when you check with `docker ps -a`. Seemed to have trouble running over the top of them?
Killing all the running containers seems to work `docker kill $(docker ps -q)`.
Then you can `docker run...` again without that error.

**Comment by tekeburak** [comment]
The following code fixed the issue.
```console
burak@docker:~$ sudo docker service restart
```

**Comment by AndiKrue** [comment]
Windows 10: after docker restart it works. but please fix it

**Comment by mushahrukhkhan** [comment]
how did you start docker in windows 10?

**Comment by ozrentk** [comment]
Yes, same error was happening to me while trying to start container with a link.
Restarting docker helped.
Docker log file is attached if anyone is interested.

MS Windows Vession 10 Pro 10.0.17763.
Docker Version 2.0.0.3 (31259).

[log.txt](https://github.com/docker/for-win/files/3269432/log.txt)

**Comment by palmin90** [comment]
same problem, fix it pls

**Comment by arianb7** [comment]
same problem,It is pain in the neck keep restarting .......

**Comment by GJStevenson** [comment]
+1

**Comment by ajitgoel** [comment]
I am able to get docker working on my windows 10 pc by resetting docker to the factory defaults.

**Comment by dragonworx** [comment]
> how did you start docker in windows 10?

@mushahrukhkhan I have Windows 10 Pro which starts Docker automatically. Otherwise I would start it manually from docker Taskbar icon.

**Comment by houliang428** [comment]
I restarted the docker and it worked. From Windows 10

**Comment by sumitgupta153** [comment]
> 
> 
> @pd93 Thanks for the informative suggestion but **I have the issue each time I stop a container**, not after restarting Windows.
> 
> I have to restart that container because it exposes 2 ports and often within minutes of being up the ports are no longer accessible (this is ibmcom/mq latest, in case anyone is interested). So the following commands give me the issue:
> `CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES 69dfe4b24676 ibmcom/mq "runmqdevserver" 8 days ago Up About a minute 0.0.0.0:1414->1414/tcp, 0.0.0.0:9443->9443/tcp, 9157/tcp pensive_mcnulty`
> `docker container stop pensive_mcnulty`
> `docker start -ai pensive_mcnulty`
> 
> > $ docker start -ai pensive_mcnulty
> > Error response from daemon: driver failed programming external connectivity on endpoint pensive_mcnulty (ffe739bcbf738391180f26b04512d1b6edefbae489df69ca6fe9daf8b6675ebe): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:9443:tcp:172.17.0.2:9443: input/output error

Right, basically that port mapping was already being used by previous running container. 
Check currently running container:  docker container ls
Stop the container that is using that port combo: `docker container stop <container ID>
Now, run: docker run -p 5001:5000 -d test-program2 (and then you can see it in browser via: localhost:5001 or 127.0.0.1:5001).

**Comment by mikeanslow** [comment]
Same issue for me, this is really annoying, any idea if there is a fix coming in for this?
Docker Desktop Version 2.0.0.3 (31259)
Windows Version 10.0.15063 Build 15063

**Comment by ajitgoel** [comment]
@endormi , you are surely right, but I was able to resolve the issue by resetting docker. restarting the system, restarting docker did not work on my development machine.

**Comment by ijustlearn** [comment]
same problem, fix it pls

**Comment by jehadassaf** [comment]
Can we please get at least a fix/wontfix on this? It's been a while since we've had to put up with this nuisance.

**Comment by jasonwilliams** [comment]
@mikeparker can you address these open issues? 

Related: https://github.com/docker/for-win/issues/1752
Related: https://github.com/docker/for-win/issues/573
Related: https://github.com/docker/for-win/issues/1038
Related: https://github.com/docker/for-win/issues/2722

Or at least close the duplicates

**Comment by johnathan-s** [comment]
yep.  It is still a problem.
Do we know that Docker and MS are aware of this one?

**Comment by mikeparker** [comment]
Hi everyone, thanks for your patience around this. We've identified the cause and will be releasing a fix with our next stable build which is imminent.

**Comment by merc-angel** [comment]
I have the same problem, after restart Docker via Docker Dekstop on Windows 10 the problem is solved.

**Comment by pythagoreantree** [comment]
I have the same problem
Error response from daemon: driver failed programming external connectivity on endpoint ...: Error starting userland proxy: /forwards/expose/port returned unexpected status: 500
Error: failed to start containers: ...

I've restarted the Docker - it doesn't help, I've restarted my PC - doesn't help.
I've uploaded new version a couple of days ago
![image](https://user-images.githubusercontent.com/40563236/62615442-ee998100-b915-11e9-90fc-a03481909661.png)

**Comment by maarten-kieft** [comment]
Well, I still face the problem? The latest release, version 2.1.0.0,  (which I have installed)  still reports the same issue. This release is from 31 july, so after the comment of Mike. I do see in the release notes that they mention something about it? errors that I get:

ERROR: for [containername] Cannot start service [containername]: driver failed programming external connectivity on endpoint [containername]             ... error                                                                             0:6379: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.

No other containers are running. Normally a restart would fix the issue, but that doesn't seem to work now. This docker-compose file always worked in the past?

**Comment by DirkWolthuis** [comment]
@maarten-kieft Same here!

**Comment by DirkWolthuis** [comment]
@maarten-kieft A restart of windows did it for me.

**Comment by maarten-kieft** [comment]
@DirkWolthuis Thanks, that seems work. I had some containers on autostart so I needed to kill first all containers, remove them, reboot my machine.  Restarting Docker in the past was earier.. 

@MikePartner: Can you have a look at the issue again? Can I provide you any information. I see clearly that if I kill all containers that still some ports are occupied by the com.docker.backened.exe process

**Comment by fgiacomelli** [comment]
Same problem for me with Docker 2.1.0.1 and windows 10 updated. Restarting Docker, and Hyper-v doesn't solve the problem. But killing all the processes related to docker and starting the Docker Desktop again does

**Comment by michaltajchertdev** [comment]
Only killing Docker.Service process helped.
Win 10 Pro, Docker 2.1.0.1
But this problem still exists from months. Until now just restart Docker Desktop was enough. Now restart Windows doesn't help. It's ridiculous.

**Comment by crgarcia12** [comment]
Since today I started having this issue. I have been using Docker on W10 for months and it never happened before

**Comment by doi155** [comment]
I also started having this issue today.
Win 10 Pro, Docker 2.1.0.1
Sometimes rebooting Windows works, but sometimes not. It is an indeterministic behavior.

```
['docker', 'run', '--name', 'dynamodb', '-d', '-p', '9000:8000', '--network', 'dynamodb-local', 'amazon/dynamodb-local']
95eafe65f3e1d97c2769ba5061d73517e6dcf02b10e4af542df14d10c09f27e6
docker: Error response from daemon: driver failed programming external connectivity on endpoint dynamodb (cd65f50495fe9bb51eedf81b884e987c915a34ba6ceae12b9d51582935951d99): Error starting userland proxy: listen tcp 0.0.0.0:9000: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
```

**Comment by shiro997** [comment]
I have this issue when i tried to create a new container on the same port of another deleted container.

Nombre del sistema operativo:              Microsoft Windows 10 Pro
Versión del sistema operativo:             10.0.17134 N/D Compilación 17134
 Version:           19.03.1

`C:\Program Files\Docker\Docker\Resources\bin\docker.exe: Error response from daemon: driver failed programming external connectivity on endpoint thirsty_newton (b0bbf2c6b84ff38bd6f144418c5b20d43e60bf79778a5d53747d3ab1158a80c8): Error starting userland proxy: listen tcp 0.0.0.0:4600: bind: Solo se permite un uso de cada dirección de socket (protocolo/dirección de red/puerto).
`

**Comment by crgarcia12** [comment]
I updated windows and docker, then restarted the laptop, and now i dont see that anymore

**Comment by jakubsgil** [comment]
> Hi everyone, thanks for your patience around this. We've identified the cause and will be releasing a fix with our next stable build which is imminent.

Hi there, I have upgraded to the newest docker version (2.1.0.1) and then started to see this issue. Restarting docker does not seem to work/resolve this. Only see this problem go away is when I reinstall docker. Is there some sort of port or firewall related issue that I might be seeing from an enterprise IT restriction?

**Comment by aldobongio** [comment]
Same problem here, Windows 10 Pro 1903, Docker Desktop 2.1.0.1.
The quickest workaround is to open the task manager, terminate "Docker.Service" and "Docker.Desktop" and finally restart the "Docker Desktop" app.

**Comment by jelordreygulle** [comment]
Restarting docker would be the solution.

**Comment by ankhuve** [comment]
> Same problem here, Windows 10 Pro 1903, Docker Desktop 2.1.0.1.
> The quickest workaround is to open the task manager, terminate "Docker.Service" and "Docker.Desktop" and finally restart the "Docker Desktop" app.

Thank you for that, I've been stuck with this issue since yesterday. Restarting Docker Desktop and Windows did not fix it for me. It seems like killing the service + docker and then starting it up again did the trick.

**Comment by jelordreygulle** [comment]
@ankhuve Good to hear.

**Comment by iagokrt** [comment]
Only killing Docker.Service process helped. + closing docker from task manager
Win 10

**Comment by harunthuo** [comment]
Adding onto the thread from all the responses from the guys above. Experienced the same issue on my Windows 10 Pro:

`listen tcp 0.0.0.0:2222: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.`

Quitting Docker Desktop and killing Docker.Service from task manager and starting Docker solved the issue

**Comment by vxhoa** [comment]
The workaround does not work for me. I am using the latest version of Docker 2.1.0.2. I already killed the process Docker.Service and re-stared docker desktop after restarting Windows 10. Docker is still showing that a container is up, I could not stop or kill the container.

**Comment by gantispam** [comment]
Same issue. 
`ERROR: for test-database  Cannot start service test-database: driver failed programming external connectivity on endpoint test-database (77ef94f584f0140f4e0b5e5e5713ab9d09a6550bfb165936153219a2f77ae4b5): Error starting userland proxy: /forwards/expose/port returned unexpected status: 500`

Restart Docker-for-windows and restart Docker container and rebuild Docker container and settings/reset/restart don't work for me.

**Comment by nogenem** [comment]
Just reinstaled Docker of Windows on a fresh Win 10 Pro yesterday and everything was working fine... I turned off the computer at night and started it again today, and now everything i get is this error while trying to start my containers... ;/ Restarting computer/docker doesn't solve the problem...

**Comment by giraffe-tree** [comment]
> > Same problem here, Windows 10 Pro 1903, Docker Desktop 2.1.0.1.
> > The quickest workaround is to open the task manager, terminate "Docker.Service" and "Docker.Desktop" and finally restart the "Docker Desktop" app.
> 
> Thank you for that, I've been stuck with this issue since yesterday. Restarting Docker Desktop and Windows did not fix it for me. It seems like killing the service + docker and then starting it up again did the trick.

it works for me =.=

**Comment by LordReaver** [comment]
Having this issue on my new Win10 Pro XPS 15, too ... however on my desktop machine, also Win10 Pro I never had this.
One time the restart worked, but today it didn't help to restart docker, quit docker, restart machine, always the same error.
I found out that somehow docker managed to keep the ports bound, although it was shut down.
Start the taskmanager and find the process "com.docker.backend.exe" - kill it and restart docker afterwards. That should release the ports and make your deployments work again.

Unluckily I couldn't reproduce this issue on purpose so I couldn't verify if this always fixes it but hopefully it's helping some of you.

**Comment by danielleiszen** [comment]
Guys this is unacceptable. This problem makes the whole containerization thing painful and slow. This technology is supposed to help us not block us. Now when I start seeing this error I need to reset to defaults download all my containers again. That takes ages. I work on different projects I should download gigs of images over and over again when it starts happening. I always turn of my computer properly, remove all the containers, quit docker desktop, despite everything it happens after a few days of usage. Please cut the "restart docker" comments and fix it ASAP.

**Comment by Gaoithe** [comment]
> > Try disabling "Experimental Features" in the Settings/Daemon menu, restarting docker and then stopping all containers.
> 
> (worked for me. from: https://stackoverflow.com/questions/49693353/error-response-from-daemon-driver-failed-programming-external-connectivity-on-e?rq=1)

Links to stackoverflow are often useful. Or educational. 

TLDR;
docker run failed for container . . . showing endpoint error . . . 
Using lsof we could see docker-proxy was still on the ip:port
Killing off that docker-proxy . . didn't help, a docker restart solved the problem. 
Conclusion: something went wrong at docker stop/docker rm time . . . docker-proxy and something else in docker endpoints was left behind.

My error showed a bind fail.

```
docker: Error response from daemon: driver failed programming external connectivity on endpoint containername (1d92a2ef61e35f97aa8d9352ff77a0d378bb73c4b0b8c46a68fdb34812ec7b17): Bind for 10.247.184.31:50000 failed: port is already allocated.
Error response from daemon: Container d20f91c0469dc3ba43e71ba12de56911418bfec915013541d4c220d11f1b6414 is not running
```

A different answer suggested using lsof to see what was causing problem. 

sudo lsof -i @ipaddress:port

```
$ sudo lsof -i @10.247.184.31:50000
COMMAND    PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
docker-pr 2900 root    4u  IPv4  43539      0t0  TCP snmp-monitoring:50000 (LISTEN)
$ ps -elf |grep 2900
4 S root       2900   1679  0  80   0 - 68205 futex_ Oct29 ?        00:00:00 /usr/bin/docker-proxy -proto tcp -host-ip 10.247.184.31 -host-port 50000 -container-ip 172.17.0.2 -container-port 50000
```

docker-proxy seems to be part of swarm network.
A kill/kill -9 resulted in defunct process, it was not showing up in lsof.

Didn't solve the problem though. :-D
Docker restart solved it.

**Comment by LoicHa** [comment]
This solved my issue on windows :

- `docker-compose down`
- Close docker desktop.
- Kill docker service
![image](https://user-images.githubusercontent.com/28562111/68081121-b554ac80-fe43-11e9-9dd5-c09d59d38304.png)
- Start docker, a popup will ask you to activate docker service via net.exe, accept
- `docker-compose up`

**Comment by shmocs** [comment]
easy to say.. but when you have 300 containers .. a docker restart will take ages and it is not a solution..
Ubuntu 18.04 here.

ERROR: for ct066  Cannot start service ct066: driver failed programming external connectivity on endpoint ct066 (4ed618d245eada38dd9ed76f550fc2cfd00c1a24bdaa830ee2dff579c669614e): Bind for 2a01:8e81:1111:aaa1::1066:11211 failed: port is already allocated

root@cts:~/nodes# lsof -i -P -n | grep 1066:11211
root@cts:~/nodes# netstat -aptunW |grep 1066:11211

nada :(

**Comment by Vespira** [comment]
> 
> 
> Guys this is unacceptable. This problem makes the whole containerization thing painful and slow. This technology is supposed to help us not block us. Now when I start seeing this error I need to reset to defaults download all my containers again. That takes ages. I work on different projects I should download gigs of images over and over again when it starts happening. I always turn off my computer properly, remove all the containers, quit docker desktop, despite everything it happens after a few days of usage. Please cut the "restart docker" comments and fix it ASAP.
> 
> And if you need more information about the problem, just tell me how to log it. I can reproduce this for sure

Exactly in the same situation. For me, restart docker is not enough, I have to Reset to factory settings, and then, I can bring up my containers. I keep up to date my Docker version, but apparently, the issue we have is different than most of the users here, because it seems to affect Docker state. I m pretty sure that the Reset action is recreating the Docker Hyper V VM instead of just restarting it, so it 'erases' the state issue.

**Comment by jacobleech** [comment]
This is still a really, really annoying issue. Even occurs running one Docker container on my Windows 10 Pro machine...

**Comment by NBardelot** [comment]
Adding myself to the long list of people impacted by this issue.

Windows up-to-date (1909), Docker up-to-date, Kubernetes up-to-date. 

I created a registry container (port 5000 mapped to the port 5000/tcp of the host). I has to kill it/recreate it, and now the port 5000/tcp is unusable. With sysinternals' TCPView I can see the process com.docker.backend.exe having that local port active (IPv4 and IPv6) :

    com.docker.backend.exe	19044	TCP	0.0.0.0	5000	0.0.0.0	0	LISTENING										
    com.docker.backend.exe	19044	TCPV6	[0:0:0:0:0:0:0:0]	5000	[0:0:0:0:0:0:0:0]	0	LISTENING

**Comment by shmocs** [comment]
#allow systemctl restart docker to keep containers running :)
```
vi /etc/docker/daemon.json
{
	"live-restore": true
}
```
then [docker service restart] will keep the containers running but will reset its shit and make 'used' ports available again

**Comment by meigallodixital** [comment]
Here too ... I am having this error a lot. I've seen this issue looking for information ...

Windows 10.0.18362 compilation 18362, Docker desktop 2.1.0.5

**Comment by NBardelot** [comment]
BTW the workaround that works for me is to kill the process 'com.docker.backend.exe' using the un-released port (using TCPView for example) and restarting Docker. It is **tedious**. About 5 to 10 minutes lost each time, depending on how many containers I have to restart.

**Comment by shmocs** [comment]
> BTW the workaround that works for me is to kill the process 'com.docker.backend.exe' using the un-released port (using TCPView for example) and restarting Docker. It is **tedious**. About 5 to 10 minutes lost each time, depending on how many containers I have to restart.

did you try my solution to restart with live-restore ?

**Comment by NBardelot** [comment]
@shmocs I'll try that the next time it breaks. Your workaround seems really nice in this situation!

**Comment by shmocs** [comment]
No idea where is that json located for windows and if it works on windows (it should work)
I have 220+ containers on a server .. and restarting containers all at once - really kills the server

**Comment by NBardelot** [comment]
On Windows the file is under `$env:userprofile/.docker` (in Powershell or `%USERPROFILE%/.docker` in a file explorer).

**Comment by meigallodixital** [comment]
In Windows Docker Desktop it is a virtualized application, not a service. When you stop it or restart it, the entire subsystem is supposedly 'cleaned' ...

I have set the parameter and comment if I notice that the problem is solved anyway :)

**Comment by aloani20** [comment]
On Windows 10, I tried the following command it solves the problem : 
C:\WINDOWS\system32>net stop winnat

The Windows NAT Driver service was stopped successfully.

you may refer to the following issue for more details 
https://github.com/docker/for-win/issues/1804

thank you

**Comment by flyke** [comment]
I have found a working solution for this error for Windows for my case.

My errors after trying lando start was:
`Starting myproject_node_1       ... error`
(the database, phpmyadmin and appserver started without problems)

`a4bcc9ce0e879a28b08be93c8a25206100dc4530e91bcfbf): Error starting userland proxy: listen tcp 0.0.0.0:3000: bind: Only one usage of each socket address`

`ERROR: for node  Cannot start service node: driver failed programming external connectivity on endpointmyproject_node_1 (bc83fa4623b66d59a4bcc9ce0e879a28b08be93c8a25206100dc4530e91bcfbf): Error starting userland proxy: listen tcp 0.0.0.0:3000: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.`

`ERROR: Encountered errors while bringing up the project.`

And that repeated a few times.
I know that restarting my pc usually fix this, but as a web developer that uses Docker and Lando everyday, I needed a solution to fix this without restarting my pc.
Closing docker and terminating the docker service did not work.
Adding "live-restore": true in C:\Users\myusername\.docker\config.json and restarting docker did not work.
Running command net stop winnat did not work.

ACTUAL WORKING SOLUTION:
error message clearly shows in this case a problem with trying to use port 3000.
So what you need to to is find whatever is hogging that port and kill it:

```
netstat -ano | findstr :3000
```

gives you a list of what exactly is using your port 3000:

```
TCP    0.0.0.0:3000       0.0.0.0:0           LISTENING         18420
TCP    [::]:3000              [::]:0                 LISTENING         18420
TCP    [::1]:3000            [::1]:62760       ESTABLISHED     18420
TCP    [::1]:62760          [::1]:3000         ESTABLISHED     15032
```

The last column is the PID (process id) that is using your port 3000. In this case: 18420 and 15032.

You can terminate those processes like this (works for me in git bash):
```
taskkill -F //PID 18420 //PID 15032
```

After this: `lando start` starts up without errors and all is working well

**Comment by AndriLaksono** [comment]
I just restart my PC (Win 10 Pro) and solved my case.

Or you can check your proxy with `netstat -ano` and see what's run. 
Kill that process.

**Comment by aprabhat** [comment]
I am getting the same error while running the docker container on docker desktop using

`docker run -p 8081:8081 {{image}}`

To check which application is using the port 8081 I used below `powershell` command

`Get-Process -Id (Get-NetTCPConnection -LocalPort 8081).OwningProcess`

which result in

> Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName
-------  ------    -----      -----     ------     --  -- -----------
    596      36  1378632      76916      39.83  29480   1 javaw

which is showing `javaw` process is running on 8081

After killing the process using 

`taskkill /F /PID 29480`

And everything is back to normal without restarting the docker desktop.

**Comment by doggy8088** [comment]
Restart `docker` service saved me.

```ps1
net stop docker
net start docker
```

**Comment by krisyupher** [comment]
add the -i option and it seems to work correctly

`docker run -d -i -p 8084:80 nginx`

**Comment by jacksonStone** [comment]
This is an issue for Docker for Mac as well. Same kind of error. Issue must be pretty gnarly if it's been around for nearly two years now. Some strange song and dance of restarting Docker a few times, killing the PID on the portt or restarting Mac is required to get things up again.

**Comment by dimat** [comment]
I had this issue on macOS. Tried to restart the computer, stopped and killed all the containers, and even did `docker system prune -a`, still nothing worked.

Since in macOS Docker is running on a virtual machine, I thought that restarting the virtual machine would help.

What I did was, just went to the Docker preferences, Resources and changed any setting (CPU or Memory). Then after applying the change it restarted the virtual machine and started working.

I guess at some point some old container didn't release all resources and was still holding the port which wasn't possible to detect from the macOS host machine.

**Comment by Pomax** [comment]
@dimat this is the repo for Docker for Windows.

**Comment by jersonmartinez** [comment]
Expose ports isn't working. For ex. apply: `docker run -dt --name app02 -p 8080:8080 appimage`

**Comment by Pomax** [comment]
@jersonmartinez that sounds like it has nothing to do with this issue: you probably want to file a [new issue](https://github.com/docker/for-win/issues/new) explaining your setup and what problem you're experiencing.

**Comment by dwivedithedev** [comment]
Restarting docker seems to work.

**Comment by Slisarenko** [comment]
/etc/init.d/redis-server stop
I'm using docker, in my case, It helped me

**Comment by krsaurabh2006** [comment]
Instead of restarting the docker you can run below commands
docker ps         /*This will list down all the processes running*/
docker stop <container id>   /*This will stop the process*/

Now you can rerun the commands

**Comment by Vespira** [comment]
I remember having this issue a long time ago with Docker for Windows old-fashionned way (Docker Engine on Hyper-V). The culprit was the fastboot of Windows 10, that messed up Docker networks when restarting computer; so you had to recreate your Docker networks to fix it.

But never had this issue again with Docker v20.10.5 on WSL2, and last response is from last year, so I guess we could close it ?
